libqtile/utils.py

```diff
@@ -52,6 +52,33 @@
 from libqtile import hook
 from libqtile.log_utils import logger
 
+# Create a list to collect references to tasks so they're not garbage collected
+# before they've run
+TASKS: list[asyncio.Task[None]] = []
+
+# Interval for removing completed tasks from list
+TASKS_INTERVAL = 10
+
+
+async def clear_tasks():
```

Comment:
Is it conceptually simpler to have our own create_task() wrapper that adds things to a list (dict?) with an additional add_done_callback() to do this removal? Then you only have to do work when it's really necessary.

Reply:
Well, yes. All of that is a much neater idea!

Will rework!